File: lingting-spring-elasticsearch/src/main/java/live/lingting/spring/elasticsearch/ElasticsearchSpringProperties.java
package live.lingting.spring.elasticsearch;

import live.lingting.framework.elasticsearch.ElasticsearchProperties;
import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.NestedConfigurationProperty;

/**
 * @author lingting 2024-03-08 14:44
 */
@Getter
@Setter
@ConfigurationProperties(ElasticsearchSpringProperties.PREFIX)
public class ElasticsearchSpringProperties {

	public static final String PREFIX = "lingting.elasticsearch";

	/**
	 * 重试配置
	 */
	@NestedConfigurationProperty
	protected ElasticsearchProperties.Retry retry = new ElasticsearchProperties.Retry();

	/**
	 * 滚动查询配置
	 */
	@NestedConfigurationProperty
	protected ElasticsearchProperties.Scroll scroll = new ElasticsearchProperties.Scroll();

	public ElasticsearchProperties properties() {
		ElasticsearchProperties properties = new ElasticsearchProperties();
		properties.setRetry(this.retry);
		properties.setScroll(this.scroll);
		return properties;
	}

}

File: lingting-spring-elasticsearch/src/main/kotlin/live/lingting/spring/elasticsearch/ElasticsearchSpringProperties.kt
package live.lingting.spring.elasticsearch

import live.lingting.framework.elasticsearch.ElasticsearchProperties
import live.lingting.framework.elasticsearch.ElasticsearchProperties.Index
import live.lingting.framework.elasticsearch.ElasticsearchProperties.Scroll
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.NestedConfigurationProperty

/**
 * @author lingting 2024-03-08 14:44
 */
@ConfigurationProperties(ElasticsearchSpringProperties.PREFIX)
class ElasticsearchSpringProperties {

    companion object {
        const val PREFIX: String = "lingting.elasticsearch"
    }

    /**
     * 索引设置
     */
    @NestedConfigurationProperty
    var index: Index = Index()

    /**
     * 重试配置
     */
    @NestedConfigurationProperty
    var retry: ElasticsearchProperties.Retry = ElasticsearchProperties.Retry()

    /**
     * 滚动查询配置
     */
    @NestedConfigurationProperty
    var scroll: Scroll = Scroll()

    fun properties(): ElasticsearchProperties {
        val properties = ElasticsearchProperties()
        properties.index = index
        properties.retry = retry
        properties.scroll = scroll
        return properties
    }

}
